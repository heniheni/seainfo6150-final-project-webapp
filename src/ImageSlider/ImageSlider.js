import React, { Component } from "react";
import styles from "./ImageSlider.module.css";

class ImagesSlide extends Component{
    constructor() {
        super();
        this.onClickNextButton = this.onClickNextButton.bind(this);
        this.onClickPreviousButton = this.onClickPreviousButton.bind(this);
        
        const image1 = require("../images/scone.PNG");
        const image2 = require("../images/sctwo.PNG");
        const image3 = require("../images/scthree.PNG");
    
        this.state = {
          index: 0,
          imageList: [image1, image2, image3],
        };
      }
    
      onClickNextButton()  {
         console.log("Forward")
        if (this.state.index + 1 === this.state.imageList.length) {
          this.setState({
            index: 0,
          });
        } else {
          this.setState({
            index: this.state.index + 1,
          });
        }
      }
    
      onClickPreviousButton() {
        console.log("backward")
        if (this.state.index - 1 === -1) {
          this.setState({
            index: this.state.imageList.length - 1,
          });
        } else {
          this.setState({
            index: this.state.index - 1,
          });
        }
      }
    
      render() {
        return (
          <div>
            <img className={styles.ImageSlider} src={this.state.imageList[this.state.index]} alt="" /> <br />
            <div className={styles.directionalButtons}>
            <button className={styles.forwardButton} onClick={this.onClickNextButton}>Next</button>
            <button className={styles.previousButton} onClick={this.onClickPreviousButton}>Previous</button>
            </div>
            
          </div>
        );
      }
}

export default ImagesSlide;