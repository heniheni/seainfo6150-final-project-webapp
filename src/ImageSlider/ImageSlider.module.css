.ImageSlider{
    max-height: 500px;
    width: 100%;  
    border-bottom: 2px solid #ef4123;
    
}

.directionalButtons{
  display: flex;
  justify-content: center;
}

.forwardButton, .previousButton {
    background-color: #19398a;
    color: white;
    height: 36px;
    width: 90px;
    border-radius: 10px;
    margin: 5px;
    font-family: Patua One, sans-serif;
    cursor: pointer;
}

