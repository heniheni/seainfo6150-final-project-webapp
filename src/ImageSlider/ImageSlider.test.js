import React from "react";
import ImageSlider from "./ImageSlider.js";

it("Renders the Image Slider component correctly", () => {
  const { container } = render(<ImageSlider />);
  expect(container).toMatchSnapshot();
});
