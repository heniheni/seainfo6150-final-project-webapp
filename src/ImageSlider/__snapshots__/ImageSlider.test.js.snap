// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Image Slider component correctly 1`] = `
<div>
  <div>
    <img
      alt=""
      class="ImageSlider"
      src="scone.PNG"
    />
     
    <br />
    <div
      class="directionalButtons"
    >
      <button
        class="previousButton"
      >
        Previous
      </button>
      <button
        class="forwardButton"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
